In [91]:
from keras.datasets import mnist

data = mnist.load_data()

In [92]:
((X_train, y_train), (X_test, y_test)) = data

In [93]:
X_train = X_train.reshape((X_train.shape[0], 28*28)).astype('float32')
X_test = X_test.reshape((X_test.shape[0], 28*28)).astype('float32')

In [94]:
X_train = X_train / 255
X_test = X_test / 255

In [95]:
from keras.utils import np_utils
print(y_test.shape)

y_train = np_utils.to_categorical(y_train)
y_test = np_utils.to_categorical(y_test)

num_classes = y_test.shape[1]
print(y_test.shape)

(10000,)
(10000, 10)


(10000,) 
(10000,10)

In [97]:
from keras.models import Sequential
from keras.layers import Dense

In [98]:
model = Sequential()
model.add(Dense(32, input_dim = 28*28, activation='relu'))
model.add(Dense(64, activation='relu'))
model.add(Dense(10, activation='softmax'))

In [100]:
model.compile(loss='categorical_crossentropy', optimizer='adam', metrics=['accuracy'])

In [101]:
model.summary()

Model: "sequential_9"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
dense_9 (Dense)              (None, 32)                25120     
_________________________________________________________________
dense_10 (Dense)             (None, 64)                2112      
_________________________________________________________________
dense_11 (Dense)             (None, 10)                650       
Total params: 27,882
Trainable params: 27,882
Non-trainable params: 0
_________________________________________________________________


model: "sequential"
_____________________________________________
Layer (type)       Output Shape    Param #
=============================================
dense (Dense)     (None, 32)      25120
_____________________________________________
dense_1 (Dense)   (None, 64)      2112
_____________________________________________
dense_2 (Dense)   (None, 10)      650
=============================================
Total params: 27,882 
Trainable params: 27882 
Non-trainable params: 0
_____________________________________________


In [102]:
model.fit(X_train, y_train, epochs=10, batch_size=100)

Epoch 1/10
600/600 [==============================] - 2s 2ms/step - loss: 0.7654 - accuracy: 0.7775
Epoch 2/10
600/600 [==============================] - 1s 2ms/step - loss: 0.2064 - accuracy: 0.9408
Epoch 3/10
600/600 [==============================] - 1s 2ms/step - loss: 0.1533 - accuracy: 0.9556
Epoch 4/10
600/600 [==============================] - 1s 2ms/step - loss: 0.1291 - accuracy: 0.9625
Epoch 5/10
600/600 [==============================] - 1s 2ms/step - loss: 0.1067 - accuracy: 0.9679
Epoch 6/10
600/600 [==============================] - 1s 2ms/step - loss: 0.0941 - accuracy: 0.9725
Epoch 7/10
600/600 [==============================] - 1s 2ms/step - loss: 0.0883 - accuracy: 0.9733
Epoch 8/10
600/600 [==============================] - 1s 2ms/step - loss: 0.0790 - accuracy: 0.9758
Epoch 9/10
600/600 [==============================] - 1s 2ms/step - loss: 0.0677 - accuracy: 0.9791
Epoch 10/10
600/600 [==============================] - 1s 2ms/step - loss: 0.0654 - accuracy: 0.9802

Epoch 1/10
600/600 [==============================] - 2s 2ms/step - loss: 0.7690 - accuracy: 0.7710
Epoch 2/10
600/600 [==============================] - 1s 2ms/step - loss: 0.2095 - accuracy: 0.9403
Epoch 3/10
600/600 [==============================] - 2s 3ms/step - loss: 0.1626 - accuracy: 0.9519
Epoch 4/10
600/600 [==============================] - 1s 2ms/step - loss: 0.1374 - accuracy: 0.9600
Epoch 5/10
600/600 [==============================] - 1s 2ms/step - loss: 0.1138 - accuracy: 0.9659
Epoch 6/10
600/600 [==============================] - 1s 2ms/step - loss: 0.1016 - accuracy: 0.9686
Epoch 7/10
600/600 [==============================] - 1s 2ms/step - loss: 0.0878 - accuracy: 0.9738
Epoch 8/10
600/600 [==============================] - 1s 2ms/step - loss: 0.0742 - accuracy: 0.9779
Epoch 9/10
600/600 [==============================] - 1s 2ms/step - loss: 0.0719 - accuracy: 0.9781
Epoch 10/10
600/600 [==============================] - 1s 2ms/step - loss: 0.0656 - accuracy: 0.9799
<tensorflow.python.keras.callbacks.history at 0x7fa231be4790>

In [103]:
scores = model.evaluate(X_test, y_test)
print(scores)

313/313 [==============================] - 0s 1ms/step - loss: 0.1024 - accuracy: 0.9703
[0.10237079858779907, 0.970300018787384]
